improver/calibration/rainforest_calibration.py

```diff
@@ -126,6 +144,433 @@ def __init__(self, model_config_dict: dict, threads: int):
                 for file in lightgbm_model_filenames
             ]
 
+    def _check_num_features(self, features: CubeList) -> None:
```

Comment:
LightGBM does this check (though it can be disabled), however treelite does not. I've tried running the code with fewer than expected features and it runs without issue when using treelite, producing sensible, but incorrect output. So I think this check is necessary, at least in the instance of treelite.